src/codegen/revert.rs

```diff
@@ -294,50 +329,57 @@ pub(super) fn revert(
         .map(|s| expression(s, cfg, contract_no, func, ns, vartab, opt));
 
     if opt.log_runtime_errors {
-        if expr.is_some() {
-            let prefix = b"runtime_error: ";
-            let error_string = format!(
-                " revert encountered in {},\n",
-                ns.loc_to_string(PathDisplay::Filename, loc)
-            );
-            let print_expr = Expression::FormatString {
-                loc: Codegen,
-                args: vec![
-                    (
-                        FormatArg::StringLiteral,
-                        Expression::BytesLiteral {
-                            loc: Codegen,
-                            ty: Type::Bytes(prefix.len() as u8),
-                            value: prefix.to_vec(),
-                        },
-                    ),
-                    (FormatArg::Default, expr.clone().unwrap()),
-                    (
-                        FormatArg::StringLiteral,
-                        Expression::BytesLiteral {
-                            loc: Codegen,
-                            ty: Type::Bytes(error_string.as_bytes().len() as u8),
-                            value: error_string.as_bytes().to_vec(),
-                        },
-                    ),
-                ],
-            };
-            cfg.add(vartab, Instr::Print { expr: print_expr });
-        } else {
-            log_runtime_error(
-                opt.log_runtime_errors,
-                "revert encountered",
-                *loc,
-                cfg,
-                vartab,
-                ns,
-            )
+        match (error_no, &expr) {
```

Comment:
There is no need to mach `error_no`. You can simplify this statement by using only a `if let Some(expr) = expr { ... } else { ... }`

Reply:
No, we only want to print for `Error(String)` which has an `expr` but no `error_no`

Reply:
You also match for `_`, which includes the case for when `error_no` is not none.

Reply:
If this was an `if let Some(expr) = expr` as you suggest. All custom errors with data would end in the `if let` branch. Which is doesn't work because they might contain data that is formattable as a string.